packages/labs/ssr/src/lib/lit-element-renderer.ts

```diff
@@ -89,11 +89,32 @@ export class LitElementRenderer extends ElementRenderer {
     const styles = (this.element.constructor as typeof LitElement)
       .elementStyles;
     if (styles !== undefined && styles.length > 0) {
-      yield '<style>';
-      for (const style of styles) {
-        yield (style as CSSResult).cssText;
+      let styleHashId: string | null = null;
```

Comment:
I think we're going to need to iterate through all the styles and emit a separate `<style-module>` for each one. That way they dedupe to an equivalent set of CSSStyleSheet objects that the components actually use.

There's even a future build optimization we could do where we rewrite the components to load styles from style cache if we know they've been SSR'ed so that we don't include them in the JS bundle when they're already on the page.

Reply:
Great comment! Can definitely do that.